tsdb/wlog/watcher.go

```diff
@@ -685,14 +685,12 @@ func (w *Watcher) readSegmentForGC(r *LiveReader, segmentNum int, _ bool) error
 			}
 			w.writer.UpdateSeriesSegment(series, segmentNum)
 
-		// Ignore these; we're only interested in series.
-		case record.Samples:
-		case record.Exemplars:
-		case record.Tombstones:
-
-		default:
+		case record.Unknown:
```

Comment:
The only minor downside that I can anticipate is that if we introduce a new type in the future that necessitates special handling here and we neglect to include it, recordsReadMetric (via a dashboard or an alert, if applicable) will no longer be capable of alerting us to the oversight. However, I believe we’ll eventually figure that out in another way.

Reply:
Yeah, I don't see any problem with defaulting to ignore records at least, since the logic is explicitly to only handle series. It's better to not reimplement `dec.Type`'s logic for detecting supported record types.

Reply:
Thinking about it some more, is it at all possible certain record types (`MmapMarkers` and `Metadata`, I assume that `HistogramSamples` and `FloatHistogramSamples` were just not added to this logic) should not be encountered here?

Reply:
@bboreham I think you wrote this code, can you clarify?

Reply:
I believe the code was copy-pasted from https://github.com/prometheus/prometheus/blob/6f595c6762d253c0ef7574b93f9795a6e3e40c81/tsdb/wlog/watcher.go#L549, with processing for non-series removed. 

I haven't been involved in work to add histograms or metadata.

More broadly, I would like the WAL to be a bit more backwards-compatible, e.g. to have a version number so we know what we're supposed to be doing, but I couldn't see a way to add that backwards-compatibly.